### PR TITLE
Add ICE config options to RTSPtoWeb

### PR DESCRIPTION
**Pull request by chriswiggins**
This PR adds in options for setting ICE servers and username/credential, pending the https://github.com/deepch/RTSPtoWeb/pull/128 PR being merged.

This should make this integration work remotely

### CODE REVIEW

File: rtsp-to-web/config.yaml [hunk 1]
```diff
@@ -21,6 +21,10 @@ options:
   http_port: 8083
   rtsp_port: 5541
   insecure_skip_verify: false
+  ice_servers:
+    - stun:stun.l.google.com:19302
+  ice_username: ""
+  ice_credential: ""
 schema:
```
Comment (allenporter) [inline rtsp-to-web/config.yaml:28]: Does the schema also need to be updated so this can show up in the UI?

File: rtsp-to-web/config.yaml [hunk 2]
```diff
@@ -21,6 +21,10 @@ options:
   http_port: 8083
   rtsp_port: 5541
   insecure_skip_verify: false
+  ice_servers:
+    - stun:stun.l.google.com:19302
+  ice_username: ""
```
Comment (allenporter) [inline rtsp-to-web/config.yaml:26]: Do you think many people will actually run their own server and this? I'm curious if we just shouldn't bother in the name of simplicity and let people that want to run their own stun server that has credentials just setup a standalone instance of this service.

Or is this actually a popular thing that people have? e.g. provided by their cloud provider or something.

File: rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web [hunk 3]
```diff
@@ -32,6 +36,9 @@ cat ${config} \
     | jq ".server.debug = ${debug}" \
     | jq ".server.log_level = \"${log_level}\"" \
     | jq ".server.http_port = \":${http_port}\"" \
+    | jq ".server.ice_servers = \":${ice_servers}\"" \
```
Comment (allenporter) [inline rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web:39]: Looks like setting ':' in front isn't quite right.  Also, i suspect the syntax for setting a value in a list using jq may be different?

**Review by allenporter — COMMENTED** [review]
Thanks for adding this. I realize this is a draft, but thought it was worth checking out anyway.

Would you also be alright with updating `DOCS.md`? I imagine some users may not love having a google stun server, but i think its reasonable to keep as a default as long as this is explained here, and we can put notes in the changelog.

Thanks again.  👍🏼

File: rtsp-to-web/config.yaml (hunk 1 above)
Comment (chriswiggins) [reply rtsp-to-web/config.yaml:28]: Valid point. I will update

File: rtsp-to-web/config.yaml (hunk 2 above)
Comment (chriswiggins) [reply rtsp-to-web/config.yaml:26]: You're probably right. Perhaps we just make it a boolean that people toggle (i.e "Use Google STUN servers for NAT Traversal") - would that work for you?

File: rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web (hunk 3 above)
Comment (chriswiggins) [reply rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web:39]: Yeah, I haven't been able to test this with the devcontainer on macOS and docker as passing through the docker socket seems to fail. I will spin up a linux VM and try that instead and test this. Thanks for the heads up

**Comment by allenporter** [comment]
I have released and updated the RTSPtoWeb binary in https://github.com/allenporter/stream-addons/pull/60 to include your fixes.

**Comment by chriswiggins** [comment]
Awesome - I've just updated this so that its a toggle to disable the Google STUN server - are you happy with that?

File: rtsp-to-web/config.yaml (hunk 2 above)
Comment (foxy82) [reply rtsp-to-web/config.yaml]: I'd say there are so many network configs out there that people might need their own STUN/TURN servers. Especially if they want to start using this with external access for things like Alexa/Google Home which would be an obvious Home Assistant use case. If we have good defaults I don't see why having the extra options hurts. @allenporter I'd prefer to use your addon than roll my own.

This morning I was trying to investigate if the random failures I get when using this as part of an Alexa doorbell are related to STUN/TURN (https://github.com/deepch/RTSPtoWeb/issues/132) so I'd be interested in these being configurable

Comment (foxy82) [reply rtsp-to-web/config.yaml]: or at the very least move the config.json to the /data area so that advanced users can work with it?

Comment (allenporter) [reply rtsp-to-web/config.yaml]: OK, lets keep the configuration options here, i'm sold.

w.r.t. /data. This is stored in /data https://github.com/allenporter/stream-addons/blob/3de0e3f19c14fb416955398482694523802c7356/rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web#L9 

Is there a doc somewhere that explains how users can manually edit configuration files in /data?

**Comment by allenporter** [comment]
> Awesome - I've just updated this so that its a toggle to disable the Google STUN server - are you happy with that?

I'm having second thoughts about over simplifying, and maybe want to go back to what you had before....

File: rtsp-to-web/config.yaml (hunk 2 above)
Comment (foxy82) [reply rtsp-to-web/config.yaml]: I think the ssh add-on and samba add-on are often mentioned but I've always run on my own OS with SSH access anyway so I've never looked into them.

File: rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web (hunk 3 above)
Comment (foxy82) [reply rtsp-to-web/rootfs/etc/cont-init.d/01-rtsp-to-web]: Correct you don't need the colon.

**Comment by chriswiggins** [comment]
OK, I've just reverted the commit 34b5ed366648750c537ff920ebe322888cd6f90a to bring it back, and have fixed the semicolon issue too

**Comment by allenporter** [comment]
Thanks, this looks great to me.

I'd like to confirm we can get this working before publishing this in a new version.  e.g. if we can say: "Remote access now works" than I think users would be more likely to accept the default google.com stun servers, but I wasn't able to get it working when configuring stun servers here and in the frontend.

Are you seeing it working end to end?

**Comment by allenporter** [comment]
I was able to get RTSPtoWeb working end to end with https://github.com/home-assistant/frontend/pull/12084 as we discussed, so this is good to merge. However, i may hold off on releasing since it can't really be used yet without the frontend fixes.

File: rtsp-to-web/config.yaml (hunk 1 above)
Comment (allenporter) [reply rtsp-to-web/config.yaml:28]: I don't think this happened? I will add this.

Comment (allenporter) [reply rtsp-to-web/config.yaml:28]: Oh I forgot, this was reverted...

**Pull request merged**